### PR TITLE
redo logging in ES6 and using google analytics as well as geordi

### PR DESCRIPTION
**Pull request by amyrebecca**
Staged: https://google-analytics.pfe-preview.zooniverse.org/

Fixes # .

Describe your changes.

# Review Checklist

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [x] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch?
- [x] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

## Optional

- [x] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [x] If changes are made to the classifier, does the dev classifier still work?
- [x] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [x] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by amyrebecca** [comment]
Just noticed that I left some things in `index.ejs` that shouldn't be there, don't merge this yet

**Comment by amyrebecca** [comment]
@wgranger just brought to my attention the other day that this hadn't been merged yet. my apologies!

**Comment by wgranger** [comment]
Before this is merged a change will need to be made to the Zooniverse Global accounts on the Google Analytics interface to account for the dimension map in `ga-log-adaptor.js`

**Comment by wgranger** [comment]
Zooniverse Global (where this is logging) is now set up with the correct custom dimensions. This will be handy when testing this PR.

File: app/lib/ga-log-adapter.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const dimensionMap = {
+  experiment: 1,
+  cohort: 2,
+  projectToken: 3,
+  userID: 4,
+  subjectID: 5,
+  workflowID: 6
```
Comment (wgranger) [inline app/lib/ga-log-adapter.js]: Does this need the extra `data` key?

Comment (amyrebecca) [reply app/lib/ga-log-adapter.js]: I just added the key while we were talking yesterday to show how easy it was--it's not actually used yet :)

File: app/partials/app.cjsx [hunk 2]
```diff
@@ -47,6 +50,7 @@ PanoptesApp = createReactClass
 
   componentWillMount: ->
     @geordiLogger = new GeordiLogger
+    @geordiLogger.subscribe(new GeordiLogAdapter(), new GALogAdapter(window.dataLayer)
```
Comment (wgranger) [inline app/partials/app.cjsx]: This needs a closing parentheses

Comment (amyrebecca) [reply app/partials/app.cjsx]: oh for goodness-- hang on one second lol

File: app/lib/ga-log-adapter.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  configure(keys) {
+    this.onRemember(keys);
+  }
+
+  onRemember(newKeys) {
+    Object.keys(newKeys).forEach((key) => {
+      if (Object.keys(dimensionMap).includes(key)) {
+        this.gaLayer('set', `dimension${dimensionMap[key]}`, newKeys[key]);
```
Comment (wgranger) [inline app/lib/ga-log-adapter.js]: It looks like this is firing when the page is loaded, and I'm getting `this.gaLayer is not a function`

Comment (amyrebecca) [reply app/lib/ga-log-adapter.js]: oh that makes sense; that will happen in places where the google analytics tracking code hasn't been installed. on production it should work fine though

Comment (wgranger) [reply app/lib/ga-log-adapter.js]: The error won't let the page load.

**Comment by wgranger** [comment]
The GA layer is found on initialization; however, gaLayer is an array, which is what causes `this.gaLayer is not a function` to appear. This is because `window.dataLayer`, that is passed into the adapter's constructor, is an array.

**Comment by amyrebecca** [comment]
It sounds like a quick workaround might be simply to include the google
analytics code after all, with the appropriate tracking bug. We might be
looking at a version incompatibility between GA and GTM; I never tested the
two in combination before. I'll be OOO until Wednesday morning so if you
need this sooner feel free to fork the PR.

**Comment by wgranger** [comment]
The most recent commit adds GA script to index.ejs. However, there are a couple concerns with the current logging.

- In order to set up GA logging based on the environment, some configuration will need to be made through Google Tag Manager.
- In this PR, events are logged to GA twice. Once through the Geordi client and once through the work on this PR. GA logging should be removed from the Geordi client.
- It looks like custom dimensions will also need to be configured through GTM. 
- Once the changes are made, the `dataLayer` object should be used instead of `window.ga`

**Comment by srallen** [comment]
@amyrebecca @wgranger It was decided in the ops channel today to just go ahead and shut down geordi because it takes up a lot of database storage. So, with that in mind, it seems the work to add Google Analytics might need some refactoring?

**Comment by amyrebecca** [comment]
this is great news. i'll try to find out what my prioritization is so i can get to this as soon as possible, or whether someone else should carry it on from here.

**Comment by amyrebecca** [comment]
@trouille this is the issue blocking the Snapshot WI google analytics stuff

**Pull request merged**